File: sample/app/src/main/java/me/shouheng/samples/stability/TestLogActivity.java
package me.shouheng.samples.stability;

import android.os.Bundle;
import android.support.annotation.Nullable;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.View;
import android.view.View.OnClickListener;
import android.widget.EditText;

import me.shouheng.samples.R;
import me.shouheng.samples.common.BaseActivity;
import me.shouheng.utils.stability.L;

import static me.shouheng.utils.ktx.LogKtxKt.loga;
import static me.shouheng.utils.ktx.LogKtxKt.logd;
import static me.shouheng.utils.ktx.LogKtxKt.loge;
import static me.shouheng.utils.ktx.LogKtxKt.logi;
import static me.shouheng.utils.ktx.LogKtxKt.logv;
import static me.shouheng.utils.ktx.LogKtxKt.logw;

public class TestLogActivity extends BaseActivity {

    private int cnt = 0;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_log_test);
        EditText et = findViewById(R.id.et);
        findViewById(R.id.btn_log).setOnClickListener(new OnClickListener() {
            @Override
            public void onClick(View v) {
                cnt++;
                switch (cnt % 6) {
                    case 0:
                        logv("Logging v");
                        break;
                    case 1:
                        logd("Logging d");
                        break;
                    case 2:
                        logi("Logging i");
                        break;
                    case 3:
                        logw("Logging w");
                        break;
                    case 4:
                        loge("Logging e");
                        break;
                    case 5:
                        loga("Logging a");
                        break;
                    default:
                        // do nothing
                }
            }
        });
        et.addTextChangedListener(new TextWatcher() {
            @Override
            public void beforeTextChanged(CharSequence s, int start, int count, int after) {
                // empty
            }

            @Override
            public void onTextChanged(CharSequence s, int start, int before, int count) {
                // empty
            }

            @Override
            public void afterTextChanged(Editable s) {
                L.d(s);
            }
        });
    }
}

File: sample/app/src/main/java/me/shouheng/samples/stability/TestLogActivity.kt
package me.shouheng.samples.stability

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.View
import android.widget.EditText
import me.shouheng.samples.R
import me.shouheng.samples.common.BaseActivity
import me.shouheng.utils.stability.L

class TestLogActivity : BaseActivity() {
    private var cnt = 0
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_log_test)
        val et = findViewById<EditText>(R.id.et)
        findViewById<View>(R.id.btn_log).setOnClickListener {
            cnt++
            when (cnt % 6) {
                0 -> L.v("Logging v")
                1 -> L.d("Logging d")
                2 -> L.i("Logging i")
                3 -> L.w("Logging w")
                4 -> L.e("Logging e")
                5 -> L.a("Logging a")
                else -> { /*noop*/ }
            }
        }
        et.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) { /*noop*/ }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) { /*noop*/ }

            override fun afterTextChanged(s: Editable) { L.d(s) }
        })
    }
}